deebot_client/message.py

```diff
@@ -190,11 +229,25 @@ def _handle_body_data(
 
         :return: A message response
         """
+        if isinstance(data, str) or isinstance(data.get("resp"), str):
+            # data = data.get('resp') if isinstance(data.get('resp', {}), str) else data
+            data = data if isinstance(data, str) else data.get("resp")
+
+            return cls._handle_body_data_xml(event_bus, str(data))
+
         if isinstance(data, dict):
             return cls._handle_body_data_dict(event_bus, data)
 
         return super()._handle_body_data(event_bus, data)
 
+    @classmethod
+    @abstractmethod
+    def _handle_body_data_xml(cls, event_bus: EventBus, data: str) -> HandlingResult:
+        """Handle message->body->data and notify the correct event subscribers.
+
+        :return: A message response
+        """
```

Comment:
Xml specific functions should be in the `messages/xml` package.

Please look at the `messages/xml/common.py`, which I implemented it for you. Do you like it or not?
If not we can try to find a solution together

Reply:
Oh definitely! I was just figuring out what would be the best way to tackle a generic XML message. 

But this is definitely a solution I like :blush: